advance-exercises/src/components/01-CompoundComponents.js

```diff
@@ -20,28 +20,48 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 class RadioGroup extends React.Component {
+  constructor(props) {
+    super(props);
+    const { defaultValue } = this.props;
+    this.state = {
+      selected: defaultValue,
+    }
+
+    this.onChange = this.onChange.bind(this);
+  }
+
   static propTypes = {
-    // defaultValue: PropTypes.string,                UN-COMMENT THIS LINE
-    children: PropTypes.shape().isRequired,
+    defaultValue: PropTypes.string,
+    children: PropTypes.arrayOf(PropTypes.shape()).isRequired,
   };
+
+  onChange(value) {
+    this.setState({
+      selected: value,
+    })
+  }
+
   render() {
-    return (
-      <div>{this.props.children}</div>
-    );
+    const { selected } = this.state;
+    const { children } = this.props;
+    return React.Children.map(children, (child) => React.cloneElement(child, { selected, onChange: this.onChange }));
```

Comment:
Use of empty lines in between variable declarations and return statement should improve readability. What do you think?